### PR TITLE
Added types for typescript

### PR DESCRIPTION
**Pull request by adamclayd**
I added an `index.d.ts` file so that you can use it in typescript. One really could use it if they are making their interface in angular

### CODE REVIEW

**Comment by GitToTheHub** [comment]
If you want get it merged, please update it against the current master. There are 3 new methods and remove the .idea-entry in .gitignore. This have nothing to do with this pr.

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    vibrate?: boolean;
+    wakeup?: boolean;
+}
+
+interface CordovaPlugins {
+    notification: {
+        local: {
+            hasPermission(callback?:  Function, scope?: Object): void;
+            requestPermission(callback?: Function, scope?: Object): void;
+            schedule(notifications: LocalNotification[], callback?: Function, scope?: Object): void;
```
Comment (anton-gustafsson) [inline types/index.d.ts:89]: The schedule function allows you to input a single object (with out a list)

```
cordova.plugins.notification.local.schedule({
    title: 'My first notification',
    text: 'Thats pretty easy...',
    foreground: true
});
```
Therefor the typing should be something like this
`schedule(notifications: LocalNotification | LocalNotification[], callback?: Function, scope?: Object): void;`

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    };
+    showWhen?: boolean;
+    silent?: boolean;
+    smallIcon?: string;
+    sound?: boolean | string;
+    sticky?: boolean;
+    summary?: string | null;
+    text?: string;
+    title?: string;
+    trigger?: {
```
Comment (anton-gustafsson) [inline types/index.d.ts:54]: When using trigger like this

```
cordova.plugins.notification.local.schedule({
    title: 'Design team meeting',
    trigger: { in: 1, unit: 'hour' }
});
```

The "at" and "every" property should be optional

**Comment by GitToTheHub** [comment]
If it's still in interest, make a new pr on the current master. Thanks.

**Pull request closed without merging**